rust/src/schema.rs

```diff
@@ -60,6 +65,132 @@ impl Field {
         // TODO Add subfields.
     }
 
+    pub fn fields(&self) -> &Vec<Field> {
+        return &self.children;
+    }
+
+    pub fn get_decoder<'a, R: Read + Seek>(
+        &'a self,
+        reader: &'a mut R,
+        page_info: PageInfo,
+    ) -> Box<dyn Decoder + '_> {
+        //Field::GetDecoder
+
+        match self.data_type().to_physical_type() {
+            Primitive(t) => match t {
+                PrimitiveType::Int8 => {
```

Comment:
hmm, these match seems very repeative, is there a way to make it shorter?

Reply:
Add boolean and null array ?

Reply:
hmm, these match seems very repeative, is there a way to make it shorter?
------
I tried but can't find a way that will help too much. I believe the key reason to make it hard is that [PrimitiveType](https://github.com/jorgecarleitao/arrow2/blob/main/src/types/mod.rs#L39) is an `enum` that we can know the value in the runtime, but the type parameter of plain decoder is a thing that we have to decide in the compile time.

Reply:
Add boolean and null array ?
------
They are not primitive types, we need to implement independent decoders for them.
For a boolean, it's a `byte` with only one valid `bit` in most programming languages, but it's actually a `bit` in arrow.
So we need converts for booleans, not just zero-copy like PlainDecoders do.
Not sure about nulls, but I think it's similar to booleans.

Reply:
We can use BooleanArray, but boolean is not an enum of [PrimitiveType](https://github.com/jorgecarleitao/arrow2/blob/main/src/types/mod.rs#L39). We can use it as another branch of this pattern matching(currently in `_ => todo!()`).